### PR TITLE
Cedar - Ana Gabriele & Citlalli Z 

### PR DESCRIPTION
**Pull request by Anagabsoares**

### CODE REVIEW

**Review by beccaelenzil — COMMENTED** [review]
🎉  Nice work!

File: app/routes.py [hunk 1]
```diff
@@ -1,2 +1,43 @@
...
+@planets_bp.route("/<planet_id>", methods=["GET"])
+def get_planet(planet_id):
+    planet_id = int(planet_id)
+    for planet in planet_lst:
+        if planet.id == planet_id:
+            return {
+                "id":planet.id,
+                "name": planet.name,
+                "type": planet.description,
+                "moons": planet.moons
```
[the remark this answers is not in the document]
Comment (Anagabsoares) [reply app/routes.py]: Thank you!!

File: app/routes.py [hunk 2]
```diff
@@ -1,2 +1,50 @@
...
+        "type": planet.type,
+        "description": planet.description,
+        "moons": planet.moons})
+    return jsonify(planets_response)
+
+@planets_bp.route("/<planet_id>", methods=["GET"])
+def get_planet(planet_id):
+    planet_id = int(planet_id)
+
+    result = [vars(planet) for planet in planet_lst if planet.id == planet_id]
```
Comment (beccaelenzil) [inline app/routes.py:46]: I just learned that `vars` will cause problems when we have a class that connects to our database. We can achieve the same effect by writing a `to_json(self)` instance method on the `Planet` class.

File: app/routes/planets_route.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+def valid_int(number, parameter_type):
+    try:
+        int(number)
+    except:
+        abort(make_response({"error": f'{parameter_type} must be an integer'}, 400))
+
+
+def get_planet_from_id(planet_id):
+    valid_int(planet_id, "planet_id")
+    return Planet.query.get_or_404(planet_id, description='{Planet not found}')
```
Comment (beccaelenzil) [inline app/routes/planets_route.py:21]: `get_or_404` is a great method to either get a instance of the planet or return a 404. In the case where it's not a `404`, we want to use the instance of the `planet` to return an appropriate response body. Here's one way we can refactor to return an appropriate response body:
```suggestion
    planet =  Planet.query.get_or_404(planet_id, description='{Planet not found}')
    return planet.to_dict()
```

File: app/routes/planets_route.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+''' Routes: planets'''
+
+@planets_bp.route("", methods =["POST"])
+def create_planets():
+    form_data = request.get_json()
+
+
+    if "title" not in form_data or "planet_type" not in form_data or "description" not in form_data or "moons" not in form_data:
+        return {"error": "incomplete request body"}, 400
```
Comment (beccaelenzil) [inline app/routes/planets_route.py:32]: Great work handling a 400 level response

File: app/routes/planets_route.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        planets = Planet.query.order_by(Planet.moons.asc())
+    elif sort_query == "desc":
+        planets = Planet.query.order_by(Planet.moons.desc())
+    else:
+        planets = Planet.query.all()
+
+    planet_response = []
+    
+    if not planets:
+        return make_response(f"Planets do not exist: Please Post some Planets into the database", 404)
```
Comment (beccaelenzil) [inline app/routes/planets_route.py:78]: This is an interesting way to handle this situation. An empty list is a little different than a True `404`. In this case, it's not that we didn't find a resource for a parameter, but it's just simply that there are no resources. It's more of a convention to return an empty array with a 200 response code.

File: app/routes/planets_route.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    for planet in planets:
+            planet_response.append(
+            planet.to_dict()
+        )
+    return jsonify(planet_response), 200
+
+
+@planets_bp.route("/<planet_id>", methods=["GET"])
+def get_planet(planet_id):
+    planet= get_planet_from_id(planet_id)
```
Comment (beccaelenzil) [inline app/routes/planets_route.py:89]: Great use of the helper function!

**Review by beccaelenzil — COMMENTED** [review]
Great work on your first Flask project. You've done a great job implementing 400 level responses, using helper functions and the `to_dict()` instance method, and using query params. I've left a few inline comments focused on design and small ways you could consider refactoring. Please let me know if you have any questions. 🎃

**Pull request closed without merging**